server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java

```diff
@@ -0,0 +1,269 @@
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.corla.endpoint;
+
+import au.org.democracydevelopers.corla.communication.requestToRaire.GenerateAssertionsRequest;
+import au.org.democracydevelopers.corla.communication.responseFromRaire.GenerateAssertionsResponse;
+import au.org.democracydevelopers.corla.util.SparkRequestStub;
+import au.org.democracydevelopers.corla.util.TestClassWithAuth;
+import au.org.democracydevelopers.corla.util.testUtils;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.google.gson.Gson;
+import org.apache.http.HttpStatus;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.ext.ScriptUtils;
+import org.testcontainers.jdbc.JdbcDatabaseDelegate;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import spark.HaltException;
+import spark.Request;
+import us.freeandfair.corla.Main;
+import us.freeandfair.corla.asm.*;
+import us.freeandfair.corla.controller.ContestCounter;
+import us.freeandfair.corla.model.AuditReason;
+import us.freeandfair.corla.model.Choice;
+import us.freeandfair.corla.model.ContestResult;
+import us.freeandfair.corla.persistence.Persistence;
+
+import javax.transaction.Transactional;
+import java.util.*;
+
+import static au.org.democracydevelopers.corla.endpoint.AbstractAllIrvEndpoint.RAIRE_URL;
+import static au.org.democracydevelopers.corla.endpoint.GenerateAssertions.CONTEST_NAME;
+import static au.org.democracydevelopers.corla.util.testUtils.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.testng.Assert.*;
+import static us.freeandfair.corla.asm.ASMEvent.DoSDashboardEvent.*;
+import static us.freeandfair.corla.endpoint.Endpoint.AuthorizationType.STATE;
+
+/**
+ * Test the GetAssertions endpoint via the API.
+ * This currently tests that the assertion generation request is accepted and blocked in the right
+ * circumstances.
+ * TODO This really isn't a completely comprehensive set of tests yet.
+ * See <a href="https://github.com/DemocracyDevelopers/colorado-rla/issues/125">...</a>
+ */
+public class GenerateAssertionsAPITests extends TestClassWithAuth {
+
+  /**
+   * Class-wide logger.
+   */
+  private static final Logger LOGGER = LogManager.getLogger(GenerateAssertionsAPITests.class);
+
+  /**
+   * Container for the mock-up database.
+   */
+  private final static PostgreSQLContainer<?> postgres = createTestContainer();
+
+  /**
+   * The Generate Assertions endpoint.
+   */
+  private final GenerateAssertions endpoint = new GenerateAssertions();
+
```

Comment:
Might be worth, for anyone reading this, to have some more detail about the parameters being supplied to the GenerateAssertionsResponse/Request parameters. eg. "Mock response for tinyExample1 contest where X is true and Y is false" or something like that. And similarly for the request, "Request for tinyExample1 contest with an allowed time of 5 seconds."